src/actions/proxy.ts

```diff
@@ -0,0 +1,161 @@
+import { Constructor, Tinlake } from '../types';
+import { waitAndReturnEvents, executeAndRetry } from '../ethereum';
+const abiCoder = require('web3-eth-abi');
+import BN from 'bn.js';
+import { ethers } from 'ethers';
```

Comment:
oh nice ethers! let's migrate entire tinlake.js to ethers <3<3<3
Otherwise would not import the library just for the address check...